eclair-core/src/test/scala/fr/acinq/eclair/channel/RestoreSpec.scala

```diff
@@ -0,0 +1,229 @@
+    val aliceListener = TestProbe()
+    channelUpdateListener.setAutoPilot(new testkit.TestActor.AutoPilot {
+      override def run(sender: ActorRef, msg: Any): TestActor.AutoPilot = msg match {
+        case u: ChannelUpdateParametersChanged if Announcements.isNode1(u.channelUpdate.channelFlags) == Announcements.isNode1(Alice.nodeParams.nodeId, Bob.nodeParams.nodeId) =>
+          aliceListener.ref.tell(msg, sender)
+          TestActor.KeepRunning
+        case _ => TestActor.KeepRunning
+      }
+    })
+    aliceListener
+  }
+
+  test("restore channel without configuration change") { f =>
+    import f._
+    val sender = TestProbe()
+    val channelUpdateListener = {
+      val listener = TestProbe()
+      system.eventStream.subscribe(listener.ref, classOf[ChannelUpdateParametersChanged])
+      aliceChannelUpdateListener(listener)
+    }
+
+    // we start by storing the current state
+    assert(alice.stateData.isInstanceOf[DATA_NORMAL])
+    val oldStateData = alice.stateData.asInstanceOf[DATA_NORMAL]
+
+    // we simulate a disconnection
+    sender.send(alice, INPUT_DISCONNECTED)
+    sender.send(bob, INPUT_DISCONNECTED)
+    awaitCond(alice.stateName == OFFLINE)
+    awaitCond(bob.stateName == OFFLINE)
+
+    // and we terminate Alice
+    alice.stop()
+
+    // we restart Alice
+    val newAlice: TestFSMRef[ChannelState, ChannelData, Channel] = TestFSMRef(new Channel(Alice.nodeParams, wallet, Bob.nodeParams.nodeId, alice2blockchain.ref, relayerA.ref, FakeTxPublisherFactory(alice2blockchain)), alicePeer.ref)
+    newAlice ! INPUT_RESTORED(oldStateData)
+
+    newAlice ! INPUT_RECONNECTED(alice2bob.ref, aliceInit, bobInit)
+    bob ! INPUT_RECONNECTED(bob2alice.ref, bobInit, aliceInit)
+    alice2bob.expectMsgType[ChannelReestablish]
+    bob2alice.expectMsgType[ChannelReestablish]
+    alice2bob.forward(bob)
+    bob2alice.forward(newAlice)
+    awaitCond(newAlice.stateName == NORMAL)
+
+    channelUpdateListener.expectNoMessage()
+  }
+
+  test("restore channel with configuration change") { f =>
+    import f._
+    val sender = TestProbe()
+    val channelUpdateListener = {
+      val listener = TestProbe()
+      system.eventStream.subscribe(listener.ref, classOf[ChannelUpdateParametersChanged])
+      aliceChannelUpdateListener(listener)
+    }
+
+    // we start by storing the current state
+    assert(alice.stateData.isInstanceOf[DATA_NORMAL])
+    val oldStateData = alice.stateData.asInstanceOf[DATA_NORMAL]
+
+    // we simulate a disconnection
+    sender.send(alice, INPUT_DISCONNECTED)
+    sender.send(bob, INPUT_DISCONNECTED)
+    awaitCond(alice.stateName == OFFLINE)
+    awaitCond(bob.stateName == OFFLINE)
+
+    // and we terminate Alice
+    alice.stop()
+
+    // we restart Alice with a different configuration
+    val newFees = RelayFees(765 msat, 2345)
+    val newConfig = Alice.nodeParams.copy(relayParams = RelayParams(newFees, newFees, newFees))
```

Comment:
nit: it would be interested to also change the `cltv_expiry_delta`

Reply:
Done with 290ec5731.

Reply:
The behavior we're interested in should happen even if only one of the parameters change. It feels weird to test if it happens when all of them change, this test would also pass if we had a `&&` instead of a `||`.

Reply:
That's a good point, I came up with 8135d11 with the goal of not making the test too complicated by testing all combinations.

Reply:
Replaced randomization by iteration over various confs in be80a5b51.